tests/api/test_soc_descriptor.cpp

```diff
@@ -342,3 +342,118 @@ TEST(SocDescriptor, CustomSocDescriptor) {
 
     EXPECT_EQ(soc_desc.get_num_dram_channels(), 1);
 }
+
+bool core_coord_vectors_equal(const std::vector<CoreCoord>& vec1, const std::vector<CoreCoord>& vec2) {
+    if (vec1.size() != vec2.size()) {
+        return false;
+    }
+
+    for (size_t i = 0; i < vec1.size(); i++) {
+        if (vec1[i] != vec2[i]) {
+            return false;
+        }
+    }
+    return true;
+}
+
+TEST(SocDescriptor, SocDescriptorGrayskullMultipleCoordinateSystems) {
+    tt_SocDescriptor soc_desc(test_utils::GetAbsPath("tests/soc_descs/grayskull_10x12.yaml"));
+
+    std::vector<CoreCoord> cores_physical = soc_desc.get_cores(CoreType::TENSIX, CoordSystem::PHYSICAL);
+
+    std::vector<CoreCoord> virtual_from_physical;
+    std::vector<CoreCoord> logical_from_physical;
+    std::vector<CoreCoord> translated_from_physical;
+
+    for (const CoreCoord& core : cores_physical) {
+        virtual_from_physical.push_back(soc_desc.translate_coord_to(core, CoordSystem::VIRTUAL));
```

Comment:
You've essentially reimplemented the get_cores completely here.

Maybe check some features of results of `get_cores` some other way, like the expected size, the CoordSystem of the returned value. Or use specific (hardcoded) expected values.